spotbugs/src/main/java/edu/umd/cs/findbugs/detect/PublicAttributes.java

```diff
@@ -0,0 +1,194 @@
+            if (field == null || writtenFields.contains(field))
+                return;
+
+            // Only consider attributes of self.
+            if (!field.getClassDescriptor().equals(getClassDescriptor()))
+                return;
+
+            if (!field.isPublic())
+                return;
+
+            bugReporter.reportBug(new BugInstance(this, "PA_PUBLIC_ATTRIBUTES",
+                    NORMAL_PRIORITY)
+                            .addClass(this).addField(field));
+            writtenFields.add(field);
+        } else if (seen == Const.AASTORE) {
+            XField field = stack.getStackItem(2).getXField();
+            // Do not report the same public field twice.
+            if (field == null || writtenFields.contains(field))
+                return;
+
+            // Only consider attributes of self.
+            if (!field.getClassDescriptor().equals(getClassDescriptor()))
+                return;
+
+            if (!field.isPublic())
+                return;
+
+            bugReporter.reportBug(new BugInstance(this, "PA_PUBLIC_ATTRIBUTES",
+                    NORMAL_PRIORITY)
+                            .addClass(this).addField(field));
+            writtenFields.add(field);
+        } else if (seen == Const.INVOKEINTERFACE ||
+                seen == Const.INVOKEVIRTUAL) {
+            XMethod xmo = getXMethodOperand();
+            if (xmo == null)
+                return;
+
+            // Heuristics: suppose that that model has a proper name in English
+            // which roughly describes its behavior, beginning with the verb.
+            // If the verb does not hint that the method writes the object
+            // then we skip it.
+            if (!looksLikeASetter(xmo.getName()))
+                return;
+
+            XField field = stack.getStackItem(xmo.getNumParams()).getXField();
+            // Do not report the same public field twice.
+            if (field == null || writtenFields.contains(field))
+                return;
+
+            // Only consider attributes of self.
+            if (!field.getClassDescriptor().equals(getClassDescriptor()))
+                return;
+
+            if (!field.isPublic() || !field.isReferenceType())
+                return;
+
+            Field clsField = getClassField(field, getClassName());
+            Type type = clsField.getType();
+            // Instances of an immutable classes cannot be modified, thus
+            // the method is not a modifier, despite its name.
+            if (type instanceof ObjectType &&
+                    isImmutable(((ObjectType) type).getClassName()))
+                return;
+
+            bugReporter.reportBug(new BugInstance(this, "PA_PUBLIC_ATTRIBUTES",
+                    NORMAL_PRIORITY)
+                            .addClass(this).addField(field));
+            writtenFields.add(field);
+        }
+    }
+
+    private static boolean isConstructor(String methodName) {
```

Comment:
This method should be renamed; it clearly covers more than just constructors.

Reply:
This conversation can be resolved, since the method got renamed in the [force-push](https://github.com/spotbugs/spotbugs/compare/4f3879c4465027dd321e347dd24a46ed4a89ccea..22a21208d56ff84c99af492aa1a20f3d66c5152d).